portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -23,6 +23,7 @@
 /** Servlet that returns some example content. TODO: modify this file to handle comments data */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
+  private ArrayList<String> comments;
```

Comment:
Try to use "interfaces" when defining member variables. In this case it is better to use a "List" instead of "ArrayList". List is an interface and ArrayList is a concrete class. In Object Oriented Programming it is better to hide "implementation details" as much as possible and in this case ArrayList is an implementation detail

Reply:
Good to know! Thanks